### ISSUE TITLE
In `preporcess.py` labels should be encoded before used.

### ISSUE THREAD

**Issue by Johnny-Wish** [Johnny-Wish/characters-recognition#11]
### Problem
From commit `d1177e23317495ffe18bcc027c79c1d7f603acc4`, it seems that labels are of shape (n_samples, 1) instead of (n_samples,), which results in classifier not working. 

### Solution
The suggested solution is to flatten the label array and use `sklearn.preprocessing.LabelEncoder` for encoding.

**Comment by Johnny-Wish** [comment]
Encoding support added.
Closing this issue.
